src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/VsProjectAdapter.cs

```diff
@@ -379,11 +379,20 @@ public async Task<string> GetNuGetLockFilePathAsync()
             return await BuildProperties.GetPropertyValueAsync(ProjectBuildProperties.NuGetLockFilePath);
         }
 
-        public async Task<bool> IsLockFileFreezeOnRestoreAsync()
+        public async Task<bool> IsRestoreLockedAsync()
         {
             await _threadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-            var value = await BuildProperties.GetPropertyValueAsync(ProjectBuildProperties.FreezeLockFileOnRestore);
+            var value = await BuildProperties.GetPropertyValueAsync(ProjectBuildProperties.RestoreLockedMode);
+
+            return MSBuildStringUtility.IsTrue(value);
+        }
+
+        public async Task<bool> IsReevaluateNuGetLockFileAsync()
```

Comment:
If I'm reading this correctly...this would allow users to overwrite perpetually correct? 
Is that something we really want in Visual Studio?

Reply:
This is to support `--reevaluate` flag with dotnet.exe this is not going to be an explicit property available to be set which is why its not documented in the spec but you can read about `--reevaluate` there. 

And just to explain, this flag is to reevaluate restore graph even if you had a lock file in order to support floating versions or ranges.

Reply:
I get that, this flag should be similar to RestoreForce. 

Then why are you reading it in Visual Studio? This is my biggest concern.

Reply:
because it will be there in visual studio as well in form of a UI gesture n i dont want two different code paths for the same option as we're doing it for force...

we can discuss this offline as well.